integration/simulation/transaction_injector.go

```diff
@@ -185,12 +185,17 @@ func (ti *TransactionInjector) issueRandomValueTransfers() {
 
 		ti.stats.Transfer()
 
-		err = obscuroClient.SendTransaction(ti.ctx, signedTx)
+		txHash, err := obscuroClient.SendTransaction(ti.ctx, signedTx)
 		if err != nil {
 			ti.logger.Info("Failed to issue transfer via RPC.", log.ErrKey, err)
 			continue
 		}
 
+		if *txHash != signedTx.Hash() {
```

Comment:
are you sure this works ? you might need to compare the bytes ( there is a util function)